### DISCUSSION TITLE
Introduce Bazel Steward - a tool for keeping dependencies up to date in Bazel

### DISCUSSION THREAD

**Discussion by lukaszwawrzyk** [apache/rocketmq#6711] [General]
Hello everyone,
I'd like to recomment using [Bazel Steward](https://github.com/VirtusLab/bazel-steward) to this repository, which simplifies the process of checking and updating dependencies in your Bazel projects, making it more efficient to keep them up-to-date. 
Bazel Steward is able to correctly update all your maven dependencies, version of Bazel itself as well as Bazel rules.

Bazel Steward automatically opens and manages pull requests. It currently only works with Github, but can be easily run from any CI. It has direct integration with Github Actions.

Check example PRs at the fork that I used for testing: https://github.com/lukaszwawrzyk/rocketmq/pulls

Any feedback is welcome.

Initial PR: https://github.com/apache/rocketmq/pull/6695

**Comment by ShadowySpirits** [comment]
> Also, could you elaborate a bit on this ASF limit? Maybe something can be done about it?

@lukaszwawrzyk Thanks! Here is a reference: [ASF GitHub action policy](https://infra.apache.org/github-actions-policy.html)

**Comment by lukaszwawrzyk** [comment]
@ShadowySpirits Ok, it seems that the major problem is that bots may not push commits to the repository. But what about dependabot that you seem to be using? What would be an acceptable workflow? How about running on a fork that opens PRs to the actual repo?
Also there is an option to run bazel-steward manually from the CLI, so it would just create branches locally on behalf of the caller and these branches could be pushed. Please also let me know if you'd be interested in this approach.
